frappe/model/base_document.py

```diff
@@ -203,11 +203,11 @@ def set(self, key, value, as_value=False):
 		if not as_value and key in self._table_fieldnames:
 			self.__dict__[key] = []
 
-			# if value is falsy, just init to an empty list
-			if value:
-				self.extend(key, value)
+			# if value is unacceptable, just init to an empty list
+			if not value or isinstance(value, str):
```

Comment:
alternatively we can disallow "all non iterable objects and string"

Reply:
we can add this later if issue arises.

Reply:
We should just let this fail? Silently handling this will suppress the bug IMO.